src/cmd/initialize.go

```diff
@@ -108,12 +113,39 @@ var initCmd = &cobra.Command{
 	},
 }
 
+func validateInitFlags() error {
+	// If 'git-url' is provided, make sure they provided values for the username and password of the push user
+	if config.InitOptions.GitServer.Address != "" {
+		if config.InitOptions.GitServer.PushUsername == "" || config.InitOptions.GitServer.PushPassword == "" {
+			return fmt.Errorf("the 'git-user' and 'git-password' flags must be provided if the 'git-url' flag is provided")
+		}
+	}
+
+	return nil
+}
+
 func init() {
 	rootCmd.AddCommand(initCmd)
 	initCmd.Flags().BoolVar(&config.CommonOptions.Confirm, "confirm", false, "Confirm the install without prompting")
 	initCmd.Flags().StringVar(&config.CommonOptions.TempDirectory, "tmpdir", "", "Specify the temporary directory to use for intermediate files")
 	initCmd.Flags().StringVar(&config.DeployOptions.Components, "components", "", "Comma-separated list of components to install.")
-	initCmd.Flags().StringVar(&config.DeployOptions.StorageClass, "storage-class", "", "Describe the StorageClass to be used")
-	initCmd.Flags().StringVar(&config.DeployOptions.Secret, "secret", "", "Root secret value that is used to 'seed' other secrets")
-	initCmd.Flags().StringVar(&config.DeployOptions.NodePort, "nodeport", "", "Nodeport to access the Zarf container registry. Between [30000-32767]")
+	initCmd.Flags().StringVar(&config.InitOptions.StorageClass, "storage-class", "", "Describe the StorageClass to be used")
+
+	// Flags for using an external Git server
+	initCmd.Flags().StringVar(&config.InitOptions.GitServer.Address, "git-url", "", "External git server url to use for this Zarf cluster")
+	initCmd.Flags().StringVar(&config.InitOptions.GitServer.PushUsername, "git-push-username", config.ZarfGitPushUser, "Username to access to the git server Zarf is configured to use. User must be able to create repositories via 'git push'")
+	initCmd.Flags().StringVar(&config.InitOptions.GitServer.PushPassword, "git-push-password", "", "Password for the push-user to access the git server")
+	initCmd.Flags().StringVar(&config.InitOptions.GitServer.ReadUsername, "git-read-username", "", "Username for read-only access to the git server")
+	initCmd.Flags().StringVar(&config.InitOptions.GitServer.ReadPassword, "git-read-password", "", "Password for the read-only user to access the git server")
```

Comment:
I think this should be standardized to `pull` instead of `read` (or rename `push` to `write`) to maintain consistency.

Reply:
I don't think the current changes will work for gitlab since the parameters in the snippet above say: `User must be able to create repositories via 'git push'`  Which is not a supported feature in Gitlab.

Reply:
That's an interesting problem we should address, would the user be expected to manually create those first, would we be on the hook to build api calls for various git servers?

Reply:
I'll test it more later but GitLab docs seem to say it's possible if your user has the right permissions. https://docs.gitlab.com/ee/user/project/working_with_projects.html#create-a-new-project-with-git-push

Reply:
Okay I see what happened in my case.  I used my personal account an generated a API token to pass to zarf-agent to do the git-operations.  It allowed the git push but it put it in my personal projects rather than the group structure that it was cloned with.

```
  DEBUG   Rewrite git URL: https://gitlab01.domain.com/devgroup/pipeline-templates.git -> https://paas-gitlab01/tworcester/mirror__paas-gitlab01.lakitu.com__devgroup__pipeline-templates.git
```